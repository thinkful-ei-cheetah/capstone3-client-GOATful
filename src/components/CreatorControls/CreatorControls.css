
.creator-page{
 display: flex;
 flex-wrap: nowrap;
}  

.aside-controls{
  width: 0;
  position: sticky;
  z-index: 11;
  top: 0;
  left: 0;
  overflow-x: hidden;
  background: linear-gradient(
    135deg,
    rgba(36, 46, 77, .9),
    rgba(137, 126, 121, .9)
  );
  color: white;
  height: 100vh;
  transition: .4s;
}

.controls-header{
  margin-top: 0px;
  padding-top: 75px;
  border-bottom: 1px solid grey;
  padding-bottom: 20px;
  text-align: center;
}

.controls-form{
  display: flex;
  flex-direction: column;
  align-items: center;
}

.control-label{
  display: block;
  margin-bottom: 10px;
}

.creator-input{
  margin-bottom: 20px;
  padding: 10px;
}

.creator-input, #creator-title {
  font-size: 16px;
  font-family: Raleway, sans-serif;
}

.creator-upload{
  margin-bottom: 20px;
}

/* Taken from https://dev.to/greduan/the-anatomy-of-a-custom-inputtypefile-component */
input[type="file"] {
  opacity: 0;
  position: absolute;
  pointer-events: none;
  /* // alternative to pointer-events, compatible with all browsers, just make it impossible to find */
  width: 1px;
  height: 1px;
}

#creator-title{
  max-width: 80%;
  padding: 8px 10px;
  width: 80%;
}


input[type="file"] + label {
  margin-top: 20px;
  display: inline-block;
  border: 1px solid grey;
  padding: 5px 10px;
  border-radius: 4px;
  margin-bottom: 30px;
  max-width: 75%;
}

input[type="file"] + label:hover{
  border: 1px solid rgb(196, 48, 43);
  cursor: pointer;
}

input[type="file"] + label:active{
  border: 1px solid rgb(196, 48, 43)
}

#creator-description{
  max-width: 80%;
  height: 150px;
  resize: none;
  width: 80%;
}

.save-preview-btn{
  display: block;
  box-shadow: 1px 1px grey;
}

.save-preview-btn:hover{
  box-shadow: none;
}

#edit-btn {
  font-size: 1.7rem;
  margin: .5rem;
  position: absolute;
  cursor: pointer;
}

@media screen and (max-width: 630px) {
  .error-msg {
    margin: 0;
    padding: 0;
  }
  
  .aside-controls {
    position: fixed;
    padding-top: 75px;
  }

  .creator-input, #creator-title {
    width: 75% !important;
  }


  .save-preview-btn {
    margin-top: 3rem;
  }

  .control-label {
    margin-top: 1em;
  }
}

@media screen and (max-width: 630px) and (max-height: 600px) {
  .controls-header {
    padding-top: 45px;
  }

  .error-msg {
    margin: 0;
    padding: 0;
  }

  input[type="file"] + label {
    margin: 0;
    margin-bottom: .75em;
  }

  #creator-description {
    max-height: 125px;
  }
}

@media screen and (min-width: 1000px) {
  #edit-form {
    width: 20%;
  }

  #edit-btn {
    visibility: hidden;
  }
}