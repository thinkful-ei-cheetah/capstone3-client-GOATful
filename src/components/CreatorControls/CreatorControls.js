import React from 'react';
import './CreatorControls.css'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faEdit, faWindowClose } from '@fortawesome/free-solid-svg-icons'

const FieldValidator = (props) => {
  if (!props.isValid) {
    return (
      <p className='error-msg'>{props.message}</p>
    )
  }
  return <></>
}

const CreatorControls = (props) => {
  const {
    handlePhoto, handleSave, handleFields, loggedIn, formValid,
    title, description, errorMessages, titleValid, descriptionValid, thumbnailValid} = props

  const handleSetOpen = (ev) => {
    ev.currentTarget.style.visibility = 'hidden';
    const editForm = document.getElementById('edit-form')
    // get screen width 
    const screenWidth = document.documentElement.clientWidth
    if (screenWidth < 630) {
      editForm.style.width = '50%'
    } else {
      editForm.style.width = '30%'
    }
  }

  const handleSetClose = (ev) => {
    const editForm = document.getElementById('edit-form')
    editForm.style.width = '0'
    const editBtn = document.getElementById('edit-btn')
    editBtn.style.visibility = 'visible'
  }

  const editFormStyles = {
    fontSize: '1.7rem',
    margin: '.5rem',
    position: 'absolute',
    cursor: 'pointer',
  }

  return (
    <>

    <div className="aside-controls" id="edit-form">
      <div className="form-heading">
        <FontAwesomeIcon 
        icon={faWindowClose} 
        className='close-btn' 
        style={editFormStyles}
        onClick={(ev) => handleSetClose(ev)}
        />
        <h3 className="controls-header">Create Preview</h3>
      </div>
      <form className="controls-form" onSubmit={handleSave}>
        <label className="control-label" htmlFor="creator-title">Title:</label>
        <FieldValidator isValid={titleValid} message={errorMessages.title}/>
        <input className="creator-input" id="creator-title" type="text" name="title" value={title} onChange={handleFields}/>
        <FieldValidator isValid={thumbnailValid} message={errorMessages.thumbnail}/>
        <input type="file" id="upload-file-btn" onChange={handlePhoto} />
        <label htmlFor="upload-file-btn">Upload Thumbnail</label>
        <label className="control-label" htmlFor="creator-description">Description:</label>
        <FieldValidator isValid={descriptionValid} message={errorMessages.description}/>
        <textarea className="creator-input" name="description" id="creator-description" value={description} onChange={handleFields} />
        <button type="submit" disabled={!formValid} className="save-preview-btn">{loggedIn ? 'Save Preview' : 'Login to save'}</button>
      </form>
    </div>
    <FontAwesomeIcon 
      icon={faEdit} 
      id='edit-btn'
      className='edit-menu-btn' 
      style={editFormStyles}
      onClick={(ev) => handleSetOpen(ev)}
    />
  </>
  ) 
}

export default CreatorControls;
