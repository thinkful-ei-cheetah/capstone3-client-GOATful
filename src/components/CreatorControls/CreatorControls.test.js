import React from 'react';
import ReactDOM from 'react-dom';
import CreatorControls from './CreatorControls';
import renderer from 'react-test-renderer';

describe('CreatorControls Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(< CreatorControls />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<CreatorControls/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
})
