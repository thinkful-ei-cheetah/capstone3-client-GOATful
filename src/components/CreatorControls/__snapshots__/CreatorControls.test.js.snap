// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatorControls Component renders the UI as expected 1`] = `
Array [
  <div
    className="aside-controls"
    id="edit-form"
  >
    <div
      className="form-heading"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-window-close fa-w-16 close-btn"
        data-icon="window-close"
        data-prefix="fas"
        focusable="false"
        onClick={[Function]}
        role="img"
        style={
          Object {
            "cursor": "pointer",
            "fontSize": "1.7rem",
            "margin": ".5rem",
            "position": "absolute",
          }
        }
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M464 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h416c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-83.6 290.5c4.8 4.8 4.8 12.6 0 17.4l-40.5 40.5c-4.8 4.8-12.6 4.8-17.4 0L256 313.3l-66.5 67.1c-4.8 4.8-12.6 4.8-17.4 0l-40.5-40.5c-4.8-4.8-4.8-12.6 0-17.4l67.1-66.5-67.1-66.5c-4.8-4.8-4.8-12.6 0-17.4l40.5-40.5c4.8-4.8 12.6-4.8 17.4 0l66.5 67.1 66.5-67.1c4.8-4.8 12.6-4.8 17.4 0l40.5 40.5c4.8 4.8 4.8 12.6 0 17.4L313.3 256l67.1 66.5z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      <h3
        className="controls-header"
      >
        Create Preview
      </h3>
    </div>
    <form
      className="controls-form"
      onSubmit={[Function]}
    >
      <label
        className="control-label"
        htmlFor="creator-title"
      >
        Title:
      </label>
      <p
        className="error-msg"
      />
      <input
        className="creator-input"
        id="creator-title"
        name="title"
        onChange={[Function]}
        type="text"
        value=""
      />
      <p
        className="error-msg"
      />
      <input
        id="upload-file-btn"
        onChange={[Function]}
        type="file"
      />
      <label
        htmlFor="upload-file-btn"
      >
        Upload Thumbnail
      </label>
      <label
        className="control-label"
        htmlFor="creator-description"
      >
        Description:
      </label>
      <p
        className="error-msg"
      />
      <textarea
        className="creator-input"
        id="creator-description"
        name="description"
        onChange={[Function]}
        value=""
      />
      <button
        className="save-preview-btn"
        disabled={true}
        type="submit"
      >
        Login to save
      </button>
    </form>
  </div>,
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-edit fa-w-18 edit-menu-btn"
    data-icon="edit"
    data-prefix="fas"
    focusable="false"
    id="edit-btn"
    onClick={[Function]}
    role="img"
    style={
      Object {
        "cursor": "pointer",
        "fontSize": "1.7rem",
        "margin": ".5rem",
        "position": "absolute",
      }
    }
    viewBox="0 0 576 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>,
]
`;
